File: update_svg_cards.py
import re
from pathlib import Path

base = Path('assets/cards')
files = list(base.glob('**/*.svg'))
if not files:
    raise SystemExit('Nenhum arquivo SVG encontrado')

font_re = re.compile(r'font-size:32px;')
mat_re = re.compile(r'transform="matrix\((-?[0-9]+\.?[0-9]*),0,0,(-?[0-9]+\.?[0-9]*),(-?[0-9]+\.?[0-9]*),(-?[0-9]+\.?[0-9]*)\)"')

changed = 0
processed = 0
for path in files:
    if 'assets/cards/back' in str(path).replace('\\', '/'):
        continue
    text = path.read_text(encoding='utf-8')
    original = text
    text = font_re.sub('font-size:36px;', text)
    if '</defs>' in text:
        head, tail = text.split('</defs>', 1)
        head = head + '</defs>'
    else:
        head = text
        tail = ''

    def repl(m):
        a = float(m.group(1))
        d = float(m.group(2))
        e = m.group(3)
        f = m.group(4)
        if abs(a) >= 1.2 and abs(a) <= 3.5 and abs(d) >= 1.2 and abs(d) <= 3.5:
            na = round(a * 0.78, 10)
            nd = round(d * 0.78, 10)
            def fmt(v):
                s = ('%.10f' % v).rstrip('0').rstrip('.')
                return s
            return f'transform="matrix({fmt(na)},0,0,{fmt(nd)},{e},{f})"'
        return m.group(0)

    tail_new = mat_re.sub(repl, tail)
    text = head + tail_new
    if text != original:
        path.write_text(text, encoding='utf-8')
        changed += 1
    processed += 1

print(f'Processed={processed} changed={changed}')

File: update_svg_cards.js
const fs = require('fs');
const path = require('path');

const baseDir = path.join(__dirname, 'assets', 'cards');
const dirs = ['clubs', 'diamonds', 'hearts', 'spades'];

const fontRegex = /font-size:32px;/g;
const matrixRegex = /transform="matrix\((-?[0-9]+\.?[0-9]*),0,0,(-?[0-9]+\.?[0-9]*),(-?[0-9]+\.?[0-9]*),(-?[0-9]+\.?[0-9]*)\)"/g;

let changedCount = 0;
let fileCount = 0;

for (const dir of dirs) {
  const folder = path.join(baseDir, dir);
  const files = fs.readdirSync(folder).filter((name) => name.endsWith('.svg'));

  for (const file of files) {
    const filePath = path.join(folder, file);
    let content = fs.readFileSync(filePath, 'utf8');
    fileCount += 1;
    const original = content;

    content = content.replace(fontRegex, 'font-size:36px;');

    let head, tail;
    const defsClose = content.indexOf('</defs>');
    if (defsClose !== -1) {
      head = content.slice(0, defsClose + 7);
      tail = content.slice(defsClose + 7);
    } else {
      const selfCloseDefs = content.search(/<defs[^>]*\/\s*>/);
      if (selfCloseDefs !== -1) {
        const match = content.match(/<defs[^>]*\/\s*>/);
        const pos = match.index + match[0].length;
        head = content.slice(0, pos);
        tail = content.slice(pos);
      } else {
        head = content;
        tail = '';
      }
    }

    tail = tail.replace(matrixRegex, (match, aStr, dStr, e, f) => {
      const a = parseFloat(aStr);
      const d = parseFloat(dStr);
      if (Math.abs(a) >= 1.2 && Math.abs(a) <= 3.5 && Math.abs(d) >= 1.2 && Math.abs(d) <= 3.5) {
        const na = +(a * 0.78).toFixed(10);
        const nd = +(d * 0.78).toFixed(10);
        const fmt = (v) => String(v).replace(/\.0+$/, '');
        return `transform="matrix(${fmt(na)},0,0,${fmt(nd)},${e},${f})"`;
      }
      return match;
    });

    const newContent = head + tail;
    if (newContent !== original) {
      fs.writeFileSync(filePath, newContent, 'utf8');
      changedCount += 1;
    }
  }
}

console.log(`Processed ${fileCount} files, changed ${changedCount} files.`);
